genie/database_to_staging.py

```diff
@@ -529,48 +601,88 @@ def mutation_in_cis_filter(
         )
-        mergeCheckDf = extract.get_syntabledf(syn=syn, query_string=query_str)
-        for center in mergeCheckDf.Center.unique():
-            if not pd.isnull(center):
-                stagingSynId = center_mappingdf.stagingSynId[
-                    center_mappingdf["center"] == center
-                ]
-                mergeCheckDf[mergeCheckDf["Center"] == center].to_csv(
-                    "mutationsInCis_filtered_samples.csv", index=False
-                )
-                load.store_file(
-                    syn=syn,
-                    filepath="mutationsInCis_filtered_samples.csv",
-                    parentid=stagingSynId[0],
-                    version_comment=genieVersion,
-                )
-                os.unlink("mutationsInCis_filtered_samples.csv")
+    remove_samples = get_mutation_in_cis_filtered_samples(
+        syn=syn, variant_filtering_synId=variant_filtering_synId
+    )
+    flagged_variants = get_mutation_in_cis_flagged_variants(
+        syn=syn, variant_filtering_synId=variant_filtering_synId
+    )
+    return (remove_samples, flagged_variants)
+
+
+def get_mutation_in_cis_filter_script_cmd(test: bool) -> str:
+    """This function gets the mutation_in_cis_filter R script
+        command call based on whether we are running in test,
+        or production mode
+
+    Args:
+        test (bool): Testing parameter.
+
+    Returns:
+        str: Full command call for the mergeCheck script
+    """
+    mergeCheck_script = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "../R/mergeCheck.R"
+    )
+    command = ["Rscript", mergeCheck_script]
+    if test:
+        command.append("--testing")
+    return command
+
+
+def get_mutation_in_cis_filtered_samples(
+    syn: synapseclient.Synapse, variant_filtering_synId: str
+) -> list:
+    """Gets the samples to remove in our variant filtering table
+       TODO: Add handling for when we have 0 row query results
+    Args:
+        syn (synapseclient.Synapse): synapse client connection
+        variant_filtering_synId (str): variant filtering synapse id
+
+    Returns:
+        pd.Series: removed samples
+    """
     query_str = (
         f"SELECT Tumor_Sample_Barcode FROM {variant_filtering_synId} where "
         "Flag = 'TOSS' and Tumor_Sample_Barcode is not null"
     )
     filtered_samplesdf = extract.get_syntabledf(syn=syn, query_string=query_str)
     # #Alex script #1 removed patients
     remove_samples = filtered_samplesdf["Tumor_Sample_Barcode"].drop_duplicates()
+    return remove_samples
+
+
+def get_mutation_in_cis_flagged_variants(
+    syn: synapseclient.Synapse, variant_filtering_synId: str
+) -> pd.Series:
+    """Gets the flagged variants in our variant filtering table which is
+       a unique string concatenation of the Chromosome, Start_Position,
+       HGVSp_Short, Reference_Allele, Tumor_Seq_Allele2 and Tumor_Sample_Barcode
+       columns
+       TODO: Add handling for when we have 0 row query results
```

Comment:
These are great TODOs to indicate weaknesses in the pipeline, but in the grand scheme of things, it is probably extremely rare moving forward that any of these tables are blank.

Reply:
Yeah this honestly should be a data quality check / expectation instead since it's directly data related. That this query should always return results.